locked-asset/energy-factory/src/unlocked_token_transfer.rs

```diff
@@ -0,0 +1,55 @@
+multiversx_sc::imports!();
+
+#[multiversx_sc::module]
+pub trait UnlockedTokenTransferModule:
+    utils::UtilsModule
+    + multiversx_sc_modules::pause::PauseModule
+    + crate::token_whitelist::TokenWhitelistModule
+{
+    #[only_owner]
+    #[endpoint(addToUnlockedTokenTransferWhitelist)]
+    fn add_to_unlocked_token_transfer_whitelist(
+        &self,
+        sc_addresses: MultiValueEncoded<ManagedAddress>,
+    ) {
+        let mapper = self.unlocked_token_transfer_whitelist();
+        for sc_addr in sc_addresses {
+            self.require_sc_address(&sc_addr);
+
+            mapper.add(&sc_addr);
+        }
+    }
+
+    #[only_owner]
+    #[endpoint(removeFromUnlockedTokenTransferWhitelist)]
+    fn remove_from_unlocked_token_transfer_whitelist(
+        &self,
+        sc_addresses: MultiValueEncoded<ManagedAddress>,
+    ) {
+        let mapper = self.unlocked_token_transfer_whitelist();
+        for sc_addr in sc_addresses {
+            mapper.remove(&sc_addr);
+        }
+    }
+
+    #[endpoint(transferUnlockedToken)]
+    fn transfer_unlocked_token(&self, dest: ManagedAddress, amount: BigUint) {
+        self.require_not_paused();
+        require!(amount != 0, "Invalid amount");
+
+        let caller = self.blockchain().get_caller();
+        require!(
+            self.unlocked_token_transfer_whitelist().contains(&caller),
+            "May not call this endpoint"
```

Comment:
Maybe add a more descriptive error msg like Not authorized to call this endpoint.